Sources/Shared/Models/PublishJob.swift

```diff
@@ -0,0 +1,88 @@
+//
+//  PublishJob.swift
+//  VimeoNetworking
+//
+//  Copyright © 2019 Vimeo. All rights reserved.
+//
+//  Permission is hereby granted, free of charge, to any person obtaining a copy
+//  of this software and associated documentation files (the "Software"), to deal
+//  in the Software without restriction, including without limitation the rights
+//  to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+//  copies of the Software, and to permit persons to whom the Software is
+//  furnished to do so, subject to the following conditions:
+//
+//  The above copyright notice and this permission notice shall be included in
+//  all copies or substantial portions of the Software.
+//
+//  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+//  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+//  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+//  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+//  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+//  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+//  THE SOFTWARE.
+//
+
+/// Data related to all supported publishing platforms, including date of first publish and a unique resource key.
+@objcMembers
+public class PublishJob: VIMModelObject {
+    
+    /// The time in ISO 8601 format when the user first published this post.
+    /// - Note: This property is available to provide interoperability with Objective-C codebases.
+    ///         Using `firstPublishDate` is preferred.
+    public private(set) var firstPublishDateString: String?
+    
+    /// The time as a `Date` when the user first published this post.
+    public private(set) var firstPublishDate: Date?
+    
+    /// The status of the post on each of the supported platforms.
+    public private(set) var destinations: PublishDestinations?
```

Comment:
I know what you mean, I think trying to figure out the naming for this stuff (and trying to distinguish between things like `PublishJobDestination` and `PublishDestination`) has been one of the most difficult parts to get a handle on.

In this case I went down the same path as you but couldn't think of anything better. I decided to leave it so that it matched what the API returned. But I'm happy to change it if we come up with anything better.

Reply:
Ok to keep as is, thanks!